examples/recognition_pipeline/providers/google_vision_api.py

```diff
@@ -0,0 +1,48 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+ 
+    https://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+from pipeline_lib.image_recognition_provider import ImageRecognitionProvider
+from google.cloud import vision_v1
+
+class GoogleVisionAPI(ImageRecognitionProvider):
+    """ An implementation of the abstract class ImageRecognitionProvider specified for Google Vision API.
+
+    """
+
+    def setup(self):
+        self.client = vision_v1.ImageAnnotatorClient()
+
+    def process(self, element):
+        features = [ {"type_": vision_v1.Feature.Type.LABEL_DETECTION} ]
+        requests = []
+        results = []
+        docs = []
+        for doc in element[1]:
+            url = doc['url']
+            image = vision_v1.Image()
+            image.source.image_uri = url
+            request = vision_v1.AnnotateImageRequest(image= image, features=features)
+            requests.append(request)
+            docs.append(doc)
+        batchRequest = vision_v1.BatchAnnotateImagesRequest(requests=requests)
+        response = self.client.batch_annotate_images(request=batchRequest)
+        for i, image_response in enumerate(response.responses):
+            allLabels = [label.description for label in image_response.label_annotations]
+            results.append([(docs[i], allLabels)])
+        return results
+    
+    get_labels = process
```

Comment:
what is this for?

Reply:
I added it just for code readability, it is not used. Should I delete it?